pkg/network/tcp_peer.go

```diff
@@ -455,3 +459,16 @@ func (p *TCPPeer) HandlePong(pong *payload.Ping) error {
 	p.lastBlockIndex = pong.LastBlockIndex
 	return nil
 }
+
+// AddGetAddrSent increments internal outstanding getaddr requests counter. The
+// peer can only send then one addr reply per getaddr request.
+func (p *TCPPeer) AddGetAddrSent() {
+	p.getAddrSent.Inc()
+}
+
+// CanProcessAddr decrements internal outstanding getaddr requests counter and
+// answers whether the addr command from the peer can be safely processed.
+func (p *TCPPeer) CanProcessAddr() bool {
+	v := p.getAddrSent.Dec()
```

Comment:
With this we can drop further `Addr` commands if there more responses has come. What about making it 0 if negative? Or setting this to at least 1 in `AddGetAddrSent`?

Reply:
We can't, it's the same as with pings, an error from `handleAddrCmd` is an error from `handleMessage` and it leads to immediate disconnection.